spec/openapi.yaml

```diff
@@ -1415,6 +1418,172 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SenderAddressRequestItem'
+        '403':
+          description: Unauthorised
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/403response'
+        '404':
+          description: Resource not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/404response'
+  /v1/messaging/numbers/sender_address/addresses/?sender_address_type=ALPHANUMERIC:
+    get:
+      tags:
+        - Source Address
+      summary: Get all approved sender address
+      operationId: GetAllApprovedSenderAddress
+      description: |
+        Get a list of approved sender addresses for your account. 
+        ```sender_address_type: ALPHANUMERIC``` only
+      parameters: []
+      responses:
+        '200':
+          description: A list of approved sender addresses for your account only
+          headers: {}
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetAllApprovedSenderAddress'
+              example:
+                data:
+                  - id: 7927d9eb-4e74-4021-836e-6cae071f84e7
+                    sender_address: EXAMPLE1
+                    sender_address_type: ALPHANUMERIC
+                    usage_type: ALPHANUMERIC
+                    destination_countries:
+                      - AU
+                    reason: 'This is my reason 1'
+                    label: 'This is my label 1'
+                    account_id: my_account,
+                    created_date: 2023-08-04T04:21:55.958Z,
+                    last_modified_date: 2023-08-04T04:21:55.958Z
+                  - id: 365dd65f-7101-46cd-8e79-e49c5620eb15
+                    sender_address: EXAMPLE2
+                    sender_address_type: ALPHANUMERIC
+                    usage_type: ALPHANUMERIC
+                    destination_countries:
+                      - AU
+                    reason: 'This is my reason 2'
+                    label: 'This is my label 2'
+                    account_id: my_account,
+                    created_date: 2023-08-14T04:21:55.958Z,
+                    last_modified_date: 2023-08-14T04:21:55.958Z
+                  - id: 4a9cb0f4-f383-40b5-84dc-bbb6a3b210dd
+                    sender_address: EXAMPLE3
+                    sender_address_type: ALPHANUMERIC
+                    usage_type: ALPHANUMERIC
+                    destination_countries:
+                      - AU
+                    reason: 'This is my reason 3'
+                    label: 'This is my label 3'
+                    account_id: my_account,
+                    created_date: 2023-08-24T04:21:55.958Z,
+                    last_modified_date: 2023-08-24T04:21:55.958Z
+                pagination:
+                  page_size: 20
+        '403':
```

Comment:
Same here, need 401 Unauthorized when auth is invalid

Reply:
Done.